core/src/io/anuke/mindustry/editor/MapRenderer.java

```diff
@@ -97,59 +101,91 @@ public void updateAll(){
     }
 
     private void render(int wx, int wy){
-        int x = wx / chunkSize, y = wy / chunkSize;
-        IndexedRenderer mesh = chunks[x][y];
-        Tile tile = editor.tiles()[wx][wy];
+        final Point windowPoint = new Point(wx, wy);
+        final Point gridPoint = new Point(wx/chunkSize, wy/chunkSize);
 
-        Team team = tile.getTeam();
-        Block floor = tile.floor();
+        final Point chunkStartPoint = getChunkStartPoint(windowPoint);
+        final int chunkArea = chunkSize * chunkSize;
+
+
+        IndexedRenderer mesh = getMesh(gridPoint);
+        Tile tile = getTile(windowPoint);
         Block wall = tile.block();
 
-        TextureRegion region;
+        int indexWall = chunkStartPoint.x + chunkStartPoint.y;
+        int indexDecal = chunkStartPoint.x  + chunkStartPoint.y + chunkArea;
 
-        int idxWall = (wx % chunkSize) + (wy % chunkSize) * chunkSize;
-        int idxDecal = (wx % chunkSize) + (wy % chunkSize) * chunkSize + chunkSize * chunkSize;
+        renderMeshRegion(windowPoint, mesh, tile, wall, indexWall);
```

Comment:
I don't see why splitting up this method is necessary at all. The new functions you're creating are only ever called from `render()` anyway; all this does is add another layer of method calls, making it more difficult to figure out what's going on.

Having lower cyclomatic complexity does not necessarily make a method more readable, especially in this case.